meshtastic/message.proto

```diff
@@ -0,0 +1,40 @@
+syntax = "proto3";
+
+package meshtastic;
+
+/*
+ * Representation of a received or sent message to be stored long-term to create a
+ * message history
+ */
+message Message {
+  /*
+   * The sending node's short name.
+   */
+  string sender_short_name = 1;
+
+ /*
+  * Whether the message is from this device.
+  */
+  bool from_self = 2;
+
+  // TODO: Comment
+  string content = 3;
+
+  /*
+   * Values 1 through 7 indicate what secondary channel index this message was sent on.
+   * Value 0 indicates that this message was sent on the primary channel. Value 8
+   * indicates that this message was sent as a direct message.
+   */
+  uint32 category = 4;
```

Comment:
What if the channel index changes over time? Also uint32 is awfully big for 8 values. Please add an option indicating this is actually uint8

Reply:
How can the channel index change?

I will change it to uint8.

Reply:
You can edit the channels on the device at any time. message for 2 weeks, then delete all channels and create different ones. The channel index is not immutable.

Reply:
To be clear on terminology, I am using channel indices and they are immutable as seen [here](https://meshtastic.org/docs/configuration/radio/channels/#channel-config-values). I would've liked to use the channel name but they may be too long to display in the UI.